src/pages/ProbeHybridisationQC.tsx

```diff
@@ -0,0 +1,391 @@
+
+  const onAddLabware = useCallback(
+    (labware: LabwareFieldsFragment) => {
+      labwares.append(labware);
+    },
+    [labwares]
+  );
+
+  const onRemoveLabware = useCallback(
+    (labware: LabwareFieldsFragment) => {
+      labwares.remove(labware.barcode);
+    },
+    [labwares]
+  );
+
+  const removeLabwareFromForm = useCallback(
+    (removeLabware, barcode: string, values: ProbeHybridisationQCFormValues) => {
+      removeLabware(barcode);
+      delete values.labwares[barcode];
+    },
+    []
+  );
+
+  const formSubmitMachine = React.useMemo(() => {
+    return createFormMachine<CompletionRequest, RecordCompletionMutation>().withConfig({
+      services: {
+        submitForm: (ctx, e) => {
+          if (e.type !== 'SUBMIT_FORM') return Promise.reject();
+          return stanCore.RecordCompletion({
+            request: e.values
+          });
+        }
+      }
+    });
+  }, [stanCore]);
+  const [currentForm, sendForm] = useMachine(() => formSubmitMachine);
+
+  const { serverError, submissionResult } = currentForm.context;
+
+  const validationWorkNumber = Yup.object().shape({
+    workNumber: Yup.string().required('SGP Number is required')
+  });
+
+  const submitForm = async (values: CompletionRequest) => sendForm({ type: 'SUBMIT_FORM', values });
+  const convertValuesAndSubmit = async (formValues: ProbeHybridisationQCFormValues) => {
+    const values: CompletionRequest = convertProbeHybridisationQCFormValuesToCompletionRequest(
+      formValues,
+      labwares.items
+    );
+    await submitForm(values);
+  };
+
+  const mapCommentOptionsToValues = useCallback((options: OptionType[]) => {
+    return options.map((v) => v.value);
+  }, []);
+
+  const updateSectionCommentsFromGlobal = useCallback(
+    (options: OptionType[], labware: LabwareFieldsFragment, values, setFieldValue) => {
+      labware.slots.forEach((slot) => {
+        slot.samples.forEach((sample) => {
+          const sampleAddressId = `${slot.address}-${sample.id}`;
+          const oldSelected = values.labwares[labware.barcode]?.sampleAddressComments?.[sampleAddressId] || [];
+          const updatedSelected = [...oldSelected, ...mapCommentOptionsToValues(options)];
+          setFieldValue(`labwares[${labware.barcode}].sampleAddressComments[${sampleAddressId}]`, updatedSelected);
+        });
+      });
+    },
+    [mapCommentOptionsToValues]
+  );
+
+  const validateCompletionDateTime = (selectedTime: string) => {
+    const selectedDateTime = new Date(selectedTime);
+    const now = new Date();
+    if (selectedDateTime > now) {
+      return 'Please select a time on or before current time';
+    }
+    return undefined;
+  };
+
+  return (
+    <AppShell>
+      <AppShell.Header>
+        <AppShell.Title>Probe Hybridisation QC</AppShell.Title>
+      </AppShell.Header>
+      <AppShell.Main>
+        <Formik<ProbeHybridisationQCFormValues>
+          initialValues={formInitialValues}
+          validationSchema={validationWorkNumber}
```

Comment:
You are not validating time field which is required

Reply:
the validation is done automatically by the input field, as I specified the min/max there